src/utils.rs

```diff
@@ -104,6 +108,94 @@ pub(crate) fn unescape_entity_to_buffer(
     ))
 }
 
+// Matching function used by unescape_xml_escapes(). It is used to match Excel
+// escape control and non-printing characters in the range '\x00' - '\x1F'.
+fn match_xml_escape(escape: &str) -> Option<&'static str> {
+    match escape {
+        "_x0000_" => Some("\x00"),
+        "_x0001_" => Some("\x01"),
+        "_x0002_" => Some("\x02"),
+        "_x0003_" => Some("\x03"),
+        "_x0004_" => Some("\x04"),
+        "_x0005_" => Some("\x05"),
+        "_x0006_" => Some("\x06"),
+        "_x0007_" => Some("\x07"),
+        "_x0008_" => Some("\x08"),
+        // No escape required for "_x0009_" = "\t"
+        // No escape required for "_x000A_" = "\n"
+        "_x000B_" => Some("\x0B"),
+        "_x000C_" => Some("\x0C"),
+        "_x000D_" => Some("\x0D"),
+        "_x000E_" => Some("\x0E"),
+        "_x000F_" => Some("\x0F"),
+        "_x0010_" => Some("\x10"),
+        "_x0011_" => Some("\x11"),
+        "_x0012_" => Some("\x12"),
+        "_x0013_" => Some("\x13"),
+        "_x0014_" => Some("\x14"),
+        "_x0015_" => Some("\x15"),
+        "_x0016_" => Some("\x16"),
+        "_x0017_" => Some("\x17"),
+        "_x0018_" => Some("\x18"),
+        "_x0019_" => Some("\x19"),
+        "_x001A_" => Some("\x1A"),
+        "_x001B_" => Some("\x1B"),
+        "_x001C_" => Some("\x1C"),
+        "_x001D_" => Some("\x1D"),
+        "_x001E_" => Some("\x1E"),
+        "_x001F_" => Some("\x1F"),
+        // Escape for literal underscore character.
+        "_x005F_" => Some("_"),
+
+        _ => None,
+    }
+}
```

Comment:
It seems to me that straitforward approach
- find `_`
- check that the next character is `x`
- find `_` again
- decode string between `x` and `_` as hex number and put a characters with such code (need to check documentation, what exactly code is used, because 4 HEX characters obviously is not enough for every unicode character) to the string

will not be more difficult, but implements a full decoding cycle. You also may not to use quick-xml decoding, but instead copy its algorithm and merge it with Excel decoding, so you won't need to allocate twice. The raw bytes you may get from the event, because it implements `AsRef<[u8]>`.